### DISCUSSION TITLE
Can't print using:  Serial.println()

### DISCUSSION THREAD

**Discussion by Stan-Reifel** [earlephilhower/arduino-pico#617] [Q&A] [answered]
I am unable to get printing working to Arduino’s console. I’m running on _Windows 10_ with Phil’s _Raspberry PI RP2040 Boards (2.0.3)_.

Using Arduino’s _Port_ command, I’ve assigned the serial port to: _COM 21 (Raspberry Pico Pi)._  
The serial port is properly showing up the _Windows Device Manager_.

Here’s my program:
```
int led_pin = 25;

void setup() 
{
  pinMode(led_pin, OUTPUT);
  Serial.begin(9600);
}

void loop() 
{
  delay(500);
  Serial.println("Starting program");

  while(1)
  {
    digitalWrite(led_pin, HIGH);   // turn the LED on (HIGH is the voltage level)
    delay(300);
    digitalWrite(led_pin, LOW);    // turn the LED off by making the voltage LOW
    delay(100);
  }
}
```

After compiling and run the program (using the Green Arrow Button) I get this message:
```
Resetting COM21
Caught exception during reset!
Converting to uf2, output size: 134144, start address: 0x2000
Flashing D: (RPI-RP2)
Wrote 134144 bytes to D:/NEW.UF2
```
The program runs and blinks the LED as expected.

If I compile again, this time with Arduino’s Serial Console window open, I get this message:
```
Resetting COM21
Caught exception during reset!
Converting to uf2, output size: 134144, start address: 0x2000
Flashing D: (RPI-RP2)
Wrote 134144 bytes to D:/NEW.UF2
processing.app.SerialException: Error opening serial port 'COM21'.
	at processing.app.Serial.<init>(Serial.java:154)
	at processing.app.Serial.<init>(Serial.java:82)
	at processing.app.SerialMonitor$2.<init>(SerialMonitor.java:132)
	at processing.app.SerialMonitor.open(SerialMonitor.java:132)
	at processing.app.AbstractMonitor.resume(AbstractMonitor.java:132)
	at processing.app.Editor.resumeOrCloseSerialMonitor(Editor.java:2156)
	at processing.app.Editor.access$1300(Editor.java:117)
	at processing.app.Editor$UploadHandler.run(Editor.java:2113)
	at java.lang.Thread.run(Thread.java:748)
Caused by: jssc.SerialPortException: Port name - COM21; Method name - openPort(); Exception type - Port not found.
	at jssc.SerialPort.openPort(SerialPort.java:167)
	at processing.app.Serial.<init>(Serial.java:143)
	... 8 more
Error opening serial port 'COM21'.

```
Again the LED blinks, but nothing is printed to the serial window. 

Any suggestions would be greatly appreciated.

**Comment by earlephilhower** [comment, marked as answer]
Because the pico serial port is actually software-driven from the pico itself
a) the upload port will disappear and a new one will appear. often at a different COM number, when an upload/reboot occurs
b) it may take several seconds before Windows recognizes the new port and sends chars to an app and so output during the first few seconds may be lost.

So, I'd add a `delay(5000)` after `Serial.begin` in your `setup`, and then check for a new port immediately after upload and change to it in the IDE

**Comment by Stan-Reifel** [comment]
Thank you Earl for taking the time to respond!

Your suggest mostly worked and is allowing me to print.  Once in a while printing still fails no matter how long I make the delay.  I've found a delay of 2000ms works just as well as 5000ms.  

I'm still getting an Exception when downloading resets the COM port, and another Exception about 50% of the time when opening it.  Occasionally I'm also getting a notification on my Windows desktop indicating problems with the serial port.  My suspicion is that the Pico's USB driver and my system are not getting along that well.

Thanks again so much for all your help!!!

**Reply by Theoristos** [reply]
Can you check this with some other terminal console program like https://sites.google.com/site/terminalbpp/ ?
And putting some print in main loop.
